pager.c

```diff
@@ -1710,8 +1710,11 @@ static int display_line(FILE *fp, LOFF_T *last_pos, struct Line **line_info,
     }
 
     /* this also prevents searching through the hidden lines */
-    if ((flags & MUTT_HIDE) && ((*line_info)[n].type == MT_COLOR_QUOTED))
+    if ((flags & MUTT_HIDE) && ((*line_info)[n].type == MT_COLOR_QUOTED) &&
+        (((*line_info)[n].quote == NULL) ||
+         ((*line_info)[n].quote->index >= C_ToggleQuotedShowLevels))) {
```

Comment:
I think this conditional is becoming too crowded and hampering readability. We should refactor it so it's easier to understand.

Something like this:
```C
struct Line cur_line_info = (*line_info)[n]; // This is probably not the right type, just using it as an example

if ((flags & MUTT_HIDE) && (cur_line_info.type == MT_COLOR_QUOTED) &&
        ((cur_line_info.quote == NULL) ||
         (cur_line_info.quote->index >= C_ToggleQuotedShowLevels)))
```
or
```C
bool should_hide(Line *line) // Probably need a different name
{
    return (line.quote == NULL) || (line.quote->index >= C_ToggleQuotedShowLevels);
}

static int display_line(...)
{
    ...
    struct Line cur_line_info = (*line_info)[n]; // This is probably not the right type, just using it as an example

    if ((flags & MUTT_HIDE) && (cur_line_info.type == MT_COLOR_QUOTED) && should_hide(cur_line_info))
    {
    ...
    }
    ...
}
```

Just to alleviate the mental burden when reading the conditional.